File: pa5/DictionaryTest.c
/* 
*  DictionaryTest.c
*  Test client for the Dictionary ADT
*/

#include<stdio.h>
#include<stdlib.h>
#include"Dictionary.h"

int main(void){
	DictionaryRef D = newDictionary();
	
	insert(D,50,12);
	insert(D,12,12);
	insert(D,34,2);
	insert(D,23,1);
        insert(D,68,15);
        insert(D,112,16);
	insert(D,45,45);
        insert(D,33,67);
        insert(D,51,78);
        insert(D,58,23);
        insert(D,5,0);
        insert(D,99,13);
	
	printDictionary(D,stdout);

	delete(D,50);
	delete(D,23);
	delete(D,99);
	
	printDictionary(D,stdout);
	
	makeEmpty(D);
	printDictionary(D,stdout);

}

File: pa3/DictionaryTest.java
/*Yugraj Singh
 * 
 *This program tests the function of the Dictionary ADT to  
 *to see if the ADT is working  
 */
class DictionaryTest{
  public static void main (String[] args){
    Dictionary A = new Dictionary();
    System.out.println(A.isEmpty());
    A.insert("4","f");
    System.out.println(A.isEmpty());
    A.insert("5","d");
    System.out.println(A.size());
    A.insert("6","3");
    A.insert("7","c");
    A.insert("3","e");
    A.insert("9","3");
   //A.delete("4");
    //A.insert("6","e"); //throws exception 
    System.out.println(A);
    System.out.println(A.lookup("7"));
    System.out.println("********");
    A.delete("6");
    System.out.println(A.size());
    System.out.println(A.lookup("6"));
    A.delete("5");
    A.delete("4");
    System.out.println(A.lookup("9"));
    System.out.println(A.size());
    System.out.println(A);
    A.makeEmpty();
    System.out.println(A.size() + " " + A.isEmpty());
    
  }
}
